tests/src/rules/max-dependencies.js

```diff
@@ -91,7 +91,7 @@ ruleTester.run('max-dependencies', rule, {
   ],
 });
 
-context('TypeScript', { skip: semver.satisfies(eslintPkg.version, '>5.0.0') }, () => {
+(semver.satisfies(tsPkg.version, '>=3.8') ? describe.skip : describe)('TypeScript', () => {
```

Comment:
Now that the tests are actually running, this predicate wasn't working either. The root reason these tests conditionally fail isn't eslint-typescript-parser or the ESLint version, it's that type-only imports were [added in TypeScript 3.8](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-8.html), and we conditionally test older versions: https://github.com/import-js/eslint-plugin-import/blob/fd85369fc219c40862d2afcfb11ad053387ba967/tests/dep-time-travel.sh#L28-L29

Predicating tests on eslint-typescript-parser or the ESLint version coincidentally works because we only install @typescript-eslint/parser on ESLint >= 5 and only downgrade TypeScript on ESLint < 3: https://github.com/import-js/eslint-plugin-import/blob/fd85369fc219c40862d2afcfb11ad053387ba967/tests/dep-time-travel.sh#L16-L18

Predicating tests directly on TypeScript 3.8 is more obvious and less error prone, I think?

I haven't investigated exhaustively, this doesn't explain the removed comment `TODO: figure out why these tests fail in eslint 4` unless it truly meant `< 4`, but the fact that type-only imports didn't exist before TypeScript 3.8 is a probable explanation?

Reply:
I think you're right, and pivoting based on the TS version makes the most sense.

Reply:
Thanks, but I was mistaken ... In fact the root reason is that type-only imports were added to @typescript-eslint/typescript-estree in https://github.com/typescript-eslint/typescript-eslint/pull/1697, whereas typescript-eslint-parser depends on [the older typescript-estree](https://unpkg.com/browse/typescript-eslint-parser@22.0.0/package.json#L49), so doesn't handle them.

eslint-plugin-import installs @typescript-eslint/parser and typescript-eslint-parser in parallel, hence @typescript-eslint/typescript-estree and typescript-estree, so I can't think of a clever way to predicate the tests on the package versions. Since typescript-eslint-parser never handles type-only imports the right thing to do is to exclude it like before, I think?

Reply:
In that case it seems fine to check the parser, yes.

Reply:
Done. Thanks!